noir-projects/aztec-nr/aztec/src/macros/notes/mod.nr

```diff
@@ -146,7 +146,10 @@ comptime fn generate_note_properties(s: StructDefinition) -> Quoted {
 
 pub(crate) comptime fn generate_note_export(s: StructDefinition, note_type_id: Field) -> Quoted {
     let name = s.name();
-    let global_export_name = f"{name}_EXPORTS".quoted_contents();
+    let mut hasher = Poseidon2Hasher::default();
+    s.as_type().hash(&mut hasher);
+    let hash = hasher.finish() as u32;
+    let global_export_name = f"{name}_{hash}_EXPORTS".quoted_contents();
```

Comment:
This is a piece of code Grego gave me to make 1 contract work with 2 notes of the same name.

Reply:
Shouldn't hurt anything, since it uses a type as input for the hash...if two contracts import the same note (exactly, not only names) should resolve to the same value!